lib/screens/v2/candidate_generator/widgets/reconstructed_alert.ex

```diff
@@ -26,26 +38,142 @@ defmodule Screens.V2.CandidateGenerator.Widgets.ReconstructedAlert do
+
+        Enum.any?(downstream_disruptions) ->
+          fullscreen_alerts =
+            find_closest_downstream_alerts(
+              downstream_disruptions,
+              stop_id,
+              location_context.stop_sequences
+            )
+
+          flex_zone_alerts = downstream_disruptions -- fullscreen_alerts
+
+          create_alert_instances(fullscreen_alerts, true, common_parameters) ++
+            create_alert_instances(flex_zone_alerts, false, common_parameters) ++
+            create_alert_instances(moderate_delays, false, common_parameters)
+
+        true ->
+          create_alert_instances(moderate_delays, true, common_parameters)
+      end
     else
       :error -> []
     end
   end
 
+  defp create_alert_instances(
+         alerts,
+         is_full_screen,
+         config: config,
+         location_context: location_context,
+         fetch_stop_name_fn: fetch_stop_name_fn,
+         is_terminal_station: is_terminal_station,
+         now: now
+       ) do
+    Enum.map(alerts, fn alert ->
+      %ReconstructedAlert{
+        screen: config,
+        alert: alert,
+        now: now,
+        location_context: location_context,
+        informed_stations_string: get_stations(alert, fetch_stop_name_fn),
+        is_terminal_station: is_terminal_station,
+        is_full_screen: is_full_screen
+      }
+    end)
+  end
+
+  defp find_closest_downstream_alerts(alerts, stop_id, stop_sequences) do
+    home_stop_distance_map = build_distance_map(stop_id, stop_sequences)
+    # Map each alert with its distance from home.
+    alerts
+    |> Enum.map(fn %{informed_entities: ies} = alert ->
+      distance =
+        ies
+        |> Enum.filter(&String.starts_with?(&1.stop, "place-"))
+        |> Enum.map(&get_distance(home_stop_distance_map, &1))
+        |> Enum.min()
+
+      {alert, distance}
+    end)
+    |> Enum.group_by(&elem(&1, 1), &elem(&1, 0))
+    |> Enum.sort_by(&elem(&1, 0))
+    # The first item will be all alerts with the shortest distance.
+    |> List.first()
+    |> elem(1)
+  end
+
+  defp build_distance_map(home_stop_id, stop_sequences) do
+    Enum.reduce(stop_sequences, %{}, fn stop_sequence, distances_by_stop ->
+      stop_sequence
+      # Index each element by its distance from home_stop_id. For example if home_stop_id is at position 2, then indices would start at -2.
+      |> Enum.with_index(-Enum.find_index(stop_sequence, &(&1 == home_stop_id)))
+      # Convert negative distances to positive, and put into a map.
+      |> Map.new(fn {stop, d} -> {stop, abs(d)} end)
+      # Merge with the distances recorded from previous stop sequences.
+      # If a stop already has a distance recorded, we use the smaller of the two. <-- *** Unsure if this is always what we'd want to do! ***
+      |> Map.merge(distances_by_stop, fn _stop, d1, d2 -> min(d1, d2) end)
+    end)
+  end
+
+  # Default to 99 if stop_id is not in distance map.
```

Comment:
Also, if we have to fall back to 99, didn't something go wrong? Should an error be logged, or is it nbd

Reply:
This could happen with delays. Because delays are just any severe delay, the home stop and the informed stop may not be connected in any stop sequence. In this case, I think it still makes sense to prioritize alerts that are reachable via the home stop.

Reply:
> delays are just any severe delay

Delays are any severe delay _on the current line_. When we fetch alerts earlier in this file, we fetch alerts that affect this line. So the home stop and informed stop should both be in the stop sequence

Reply:
See [here](https://github.com/mbta/screens/pull/1765#discussion_r1219858317) for another reason I think a fallback is good to have.

Reply:
To respond to that PS here: Gov't Center _does_ have a relationship to Kenmore, even if the alert we're looking at is only on the E branch. For Govt Center, the `home_stop_distance_map` includes the route sequence for all the routes that go through it. 

The problem is when we have a home stop on the branch, like Ashmont. Then an alert on the _other_ branch does cause the default of 99 to kick in, since the Ashmont route sequence doesn't have the opposite branch stations. So